plugins/outputs/kafka/kafka.go

```diff
@@ -209,6 +210,15 @@ func (k *Kafka) Write(metrics []telegraf.Metric) error {
 			Value: sarama.ByteEncoder(buf),
 		}
 
+		if k.MetricNameHeader != "" {
+			m.Headers = []sarama.RecordHeader{
+				{
+					Key:   []byte(k.MetricNameHeader),
+					Value: []byte(metric.Name()),
+				},
+			}
+		}
```

Comment:
Would it make sense to initialize the header list once in the `Init()` function and store it to then just use it here?

Reply:
Initialize header list and override value here or how to initilize header list when value is known only here 🤔

Reply:
I might be missing something but in `Init()` you could do
```golang
	if k.MetricNameHeader != "" {
		k.recordHeaders = []sarama.RecordHeader{
			{
				Key:   []byte(k.MetricNameHeader),
				Value: []byte(metric.Name()),
			},
		}
	}
````
and the here you do
```suggestion
		m.Headers = k,recordHeaders
```
Can't you?

Reply:
Can you access `metric.Name()` in Init() method to populate header value? I thought it's not possible

Reply:
:facepalm: sorry I completely missed this! Sorry for the fuzz and of course your code is correct!